### PR TITLE
Use <button type=submit> instead of <input>

### PR DESCRIPTION
**Pull request by meshy**

### CODE REVIEW

**Comment by meshy** [comment]
@incuna/backend please review

File: groups/tests/test_forms.py [hunk 1]
```diff
@@ -26,6 +28,28 @@ def test_form_not_valid(self):
         self.assertFalse(form.is_valid())
         self.assertIn('body', form.errors)
 
+    def test_submit_not_input(self):
+        """The form does not have a submit <input>."""
+        rendered_form = render_crispy_form(self.form())
+
+        inputs = BeautifulSoup(rendered_form, 'html.parser').findAll('input')
+
+        has_submit = any(filter(lambda i: i.get('type') == 'submit', inputs))
```
Comment (Ian-Foote) [inline groups/tests/test_forms.py]: I think I prefer:

```python
has_submit = any(i for i in inputs if i.get('type') == 'submit')
```

Comment (meshy) [inline groups/tests/test_forms.py]: :+1:

Comment (meshy) [inline groups/tests/test_forms.py]: 5bdeb89

Comment (Ian-Foote) [inline groups/tests/test_forms.py]: There's multiple tests you have the same expression in. :wink:

Comment (meshy) [inline groups/tests/test_forms.py]: Rebased --  a097492

**Comment by meshy** [comment]
You know -- I'm probably going a little whitespace-crazy on this. I'm going to go cull a load of them.

File: groups/forms.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def build_helper(self):
         helper.layout = Layout(
             'body',
             FormActions(
-                Submit('comment-submit', 'Post comment'),
+                StrictButton('Post comment', type='submit'),
```
Comment (KevinEtienne) [inline groups/forms.py]: Should `'Post comment'` be translatable?

File: groups/tests/test_forms.py [hunk 3]
```diff
@@ -70,6 +104,23 @@ def test_form_not_valid(self):
         self.assertFalse(form.is_valid())
         self.assertIn('name', form.errors)
 
+    def test_submit_not_input(self):
+        """The form does not have a submit <input>."""
+        rendered_form = render_crispy_form(self.form())
+        inputs = BeautifulSoup(rendered_form, 'html.parser').findAll('input')
```
Comment (KevinEtienne) [inline groups/tests/test_forms.py]: can you do 
```
inputs = BeautifulSoup(rendered_form, 'html.parser').findAll('input', {'type':'submit'})
self.assertEqual(inputs, [])
```

File: groups/forms.py (hunk 2 above)
Comment (adam-thomas) [inline groups/forms.py]: ^ This, same with the other strings in this file.

File: groups/tests/test_forms.py (hunk 3 above)
Comment (Ian-Foote) [inline groups/tests/test_forms.py]: If that works, that's certainly nicer.

File: groups/tests/test_forms.py [hunk 4]
```diff
@@ -88,17 +139,43 @@ def get_form(self, **kwargs):
...
+        """The button says 'Subscribe' when the user is not subscribed."""
+        rendered_form = render_crispy_form(self.get_form())
+        button = BeautifulSoup(rendered_form, 'html.parser').findAll('button')[0]
+        self.assertEqual(button.string, 'Subscribe')
+
+    def test_button_text_unsubscribe(self):
+        """The button says 'Unsubscribe' when the user is subscribed."""
+        self.discussion.subscribers.add(self.user)
+        rendered_form = render_crispy_form(self.get_form())
+        button = BeautifulSoup(rendered_form, 'html.parser').findAll('button')[0]
```
Comment (adam-thomas) [inline groups/tests/test_forms.py]: Is there a `find('button')` method that'll do the same thing as `findAll('button')[0]`?

File: groups/forms.py (hunk 2 above)
Comment (meshy) [inline groups/forms.py]: Yes it should -- but that's one for another PR, I think, as there will surely be other places that need fixing too

**Comment by meshy** [comment]
I've refactored the tests to remove a little boilerplate -- sorry, it seems to have clobbered some comments

File: groups/forms.py (hunk 2 above)
Comment (adam-thomas) [inline groups/forms.py]: fair :)

File: groups/tests/test_forms.py (hunk 3 above)
Comment (meshy) [inline groups/tests/test_forms.py]: :+1: I'll use something like that, thanks `:)`

File: groups/forms.py (hunk 2 above)
Comment (adam-thomas) [inline groups/forms.py]: https://github.com/incuna/incuna-groups/issues/30

File: groups/tests/test_forms.py (hunk 4 above)
Comment (meshy) [inline groups/tests/test_forms.py]: There is `findNext`, but I didn't get that to work properly, unfortunately `:(`

File: groups/forms.py (hunk 2 above)
Comment (meshy) [inline groups/forms.py]: Thanks!

**Comment by meshy** [comment]
Updated

**Comment by adam-thomas** [comment]
Needs master, otherwise LGTM, sorry! :)

File: changelog.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+Unreleased
+----------
+
+* Use html `buttons` instead of `inputs` for form submission.
```
Comment (adam-thomas) [inline changelog.md]: It occurs to me that this might be a `v2.0.0` on release, since it might be backwards-incompatible for some CSS styling setups.

**Comment by adam-thomas** [comment]
:+1:

**Comment by meshy** [comment]
Thanks for the reviews `:)`

File: changelog.md (hunk 5 above)
Comment (meshy) [inline changelog.md]: I was thinking the same thing.

**Pull request merged**